tools/config/src/fullnode_config.rs

```diff
@@ -0,0 +1,174 @@
+use diem_config::config::NodeConfig;
+use diem_types::{network_address::NetworkAddress, waypoint::Waypoint, PeerId};
+use libra_types::legacy_types::app_cfg::default_file_path;
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+/// fetch seed peers and make a yaml file from template
+pub async fn init_fullnode_yaml(home_dir: Option<PathBuf>) -> anyhow::Result<()> {
+    let waypoint = get_genesis_waypoint(home_dir.clone()).await?;
+
+    let yaml = make_fullnode_yaml(home_dir.clone(), waypoint)?;
+
+    let home = home_dir.unwrap_or_else(default_file_path);
+    let p = home.join("fullnode.yaml");
+
+    std::fs::write(&p, yaml)?;
+    let peers = fetch_seed_addresses(None).await?;
+
+    add_peers_to_yaml(&p, peers)?;
+
+    Ok(())
+}
+
+pub fn add_peers_to_yaml(
+    path: &Path,
+    peers: HashMap<PeerId, Vec<NetworkAddress>>,
+) -> anyhow::Result<()> {
+    let string = std::fs::read_to_string(path)?;
+    let mut parsed: NodeConfig = serde_yaml::from_str(&string)?;
+
+    parsed.full_node_networks.iter_mut().for_each(move |e| {
+        if e.network_id.is_public_network() {
+            e.seed_addrs = peers.clone();
+        }
+    });
+
+    let ser = serde_yaml::to_string(&parsed)?;
+    std::fs::write(path, ser)?;
+
+    Ok(())
+}
+
+/// get seed peers from an upstream url
+pub async fn fetch_seed_addresses(
+    url: Option<&str>,
+) -> anyhow::Result<HashMap<PeerId, Vec<NetworkAddress>>> {
+    let u = url.unwrap_or(
+        "https://raw.githubusercontent.com/0LNetworkCommunity/seed-peers/main/seed_peers.yaml",
+    );
+    let res = reqwest::get(u).await?;
+    let out = res.text().await?;
+    let seeds: HashMap<PeerId, Vec<NetworkAddress>> = serde_yaml::from_str(&out)?;
+
+    Ok(seeds)
+}
+
+/// Create a fullnode yaml to bootstrap node
+pub fn make_fullnode_yaml(home_dir: Option<PathBuf>, waypoint: Waypoint) -> anyhow::Result<String> {
+    let home_dir = home_dir.unwrap_or_else(libra_types::global_config_dir);
+    let path = home_dir.display().to_string();
+
+    let template = format!(
+        "
+base:
+  role: 'full_node'
+  data_dir: '{path}/data'
+  waypoint:
+    from_config: '{waypoint}'
+
+execution:
+  genesis_file_location: '{path}/genesis.blob'
+
+state_sync:
+     state_sync_driver:
+        bootstrapping_mode: DownloadLatestStates
+        continuous_syncing_mode: ExecuteTransactionsOrApplyOutputs
+
+full_node_networks:
+- network_id: 'public'
+  listen_address: '/ip4/0.0.0.0/tcp/6182'
+
+api:
+  enabled: true
+  address: '0.0.0.0:8080'
+"
+    );
+    Ok(template)
+}
+
+/// download genesis blob
+pub async fn download_genesis(home_dir: Option<PathBuf>) -> anyhow::Result<()> {
+    let bytes = reqwest::get("https://github.com/0xzoz/blob/raw/main/genesis.blob")
```

Comment:
Should we have repositories stored in the 0LNetworkCommunity repo for this? Seperate repos for REX and Mainnet

Reply:
yeah I have this as a placeholder. 
Perhaps keep it in the original network genesis repo, under a directory: `upgrades`